packages/communication-react/review/beta/communication-react.api.md

```diff
@@ -132,6 +132,35 @@ export type AreParamEqual<A extends (props: any) => JSX.Element | undefined, B e
 // @public
 export type AreTypeEqual<A, B> = A extends B ? (B extends A ? true : false) : false;
 
+// @beta
+export interface AtMentionDisplayOptions {
+    atMentionSuggestionRenderer?: (suggestion: AtMentionSuggestion) => JSX.Element;
+}
+
+// @internal
+export interface _AtMentionFlyoutProps {
+    onDismiss?: () => void;
+    onSuggestionSelected?: (suggestion: AtMentionSuggestion) => void;
+    suggestionItemRenderer?: (suggestion: AtMentionSuggestion, onSuggestionSelected?: (suggestion: AtMentionSuggestion) => void) => JSX.Element;
+    suggestions: AtMentionSuggestion[];
+    target?: React_2.RefObject<Element>;
+    title?: string;
+}
+
+// @beta
+export interface AtMentionLookupOptions {
```

Comment:
Any special reason that we separate AtMentionDisplayOptions and AtMentionLookupOptions? it feels that you could merge them into one single option object like (AtMentionOptions) and make them optional. The idea is to reduce the props burden of messagethread

Reply:
@PorterNan Display option is used for displaying mentions while lookup option is used when sending/editing messages. MessageThread happens to need both because both displaying and editing goes through it. I think it would be a bit confusing if we combine them.

Reply:
For AtMention feature itself, it makes sense to have it separated, but if you look at from a higher layer, it is just all AtMention feature related options, if people feel confused then that means we should give the property a better name

Reply:
@PorterNan  The two options can potentially grow and combine them will make it hard for Contoso developers to figure out which ones they need for one of these 2 features. For example, for sendBox, we only need AtMentionLookupOptions, it's very easy and clear with  AtMentionLookupOptions  since all the necessary options are defined together and it's more type guarded. While as if we combine them, developers need to figure out which ones are for the lookup features for sendBox among the combined options.

Reply:
> Any special reason that we separate AtMentionDisplayOptions and AtMentionLookupOptions? it feels that you could merge them into one single option object like (AtMentionOptions) and make them optional. The idea is to reduce the props burden of messagethread

I think what you're really asking for here is to have one `MentionOptions` optional object on `MessageThreadProps` that contains optional `MentionDisplayOptions` and `MentionLookupOptions`.
The two sub types are definitely separate sets of usage, but I see we would want to keep the `MessageThreadProps` as simple as possible.

Reply:
@Leah-Xia-Microsoft, it's no doubt that sendBox just need one of the option to make it work, but we can define a single prop for sendbox instead of adding too many options to message thread
You can choose to do type MentionOptions = MentionDisplayOptions & MentionLookupOptions to make it work in messageThread

Also @emlynmac's way is a good one, we just need to give those nested options a good property name